packages/xdl/src/Project.ts

```diff
@@ -1847,15 +2019,15 @@ export async function stopExpoServerAsync(projectRoot: string) {
 
 async function _connectToNgrokAsync(
   projectRoot: string,
-  args: mixed,
-  hostnameAsync: Function,
-  ngrokPid: ?number,
+  args: ngrok.NgrokOptions,
+  hostnameAsync: () => Promise<string>,
+  ngrokPid: number | null | undefined,
```

Comment:
Might be worth looking at the caller and seeing if we can consistently pass in null or undefined but not both.

Reply:
The value comes from `ProjectSettings.readPackagerInfoAsync` which has this type:
```ts
type PackagerInfo = {
  expoServerPort?: number | null;
  packagerPort?: number | null;
  packagerPid?: number | null;
  expoServerNgrokUrl?: string | null;
  packagerNgrokUrl?: string | null;
  ngrokPid?: number | null;
  devToolsPort?: number | null;
  webpackServerPort?: number | null;
};
```
Initially all of these properties are non-existent (`undefined`) and then they are set to numbers/string as these services get started. Then when the services are stopped the ports/PIDs are explicitly set to `null` to mark the service as stopped. We treat `undefined` and `null` the same.

I agree setting the PIDs/ports to `undefined` (=removing the field from JSON) would be more ideal and clean. But we should also keep it consistent so that if we set `ngrokPid` to `undefined`, we also treat `packagerPid` or `expoServerPort` the same way.

Is it worth it to go over every call that sets any of these values to `null` and change it to `undefined`?

Reply:
Let's just accept `null | undefined` for now. In this case, pid = 0 is reserved anyway so if we just do a truthiness check for the pid it will always be non-zero if set.